debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1195,28 +1195,32 @@ private int checkJ9Object(J9ObjectPointer object, GCHeapRegionDescriptor regionD
 			}
 		}
 
-		if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
-
-			boolean isCorrectData = true;
-			J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
-			try {
-				if (J9IndexableObjectHelper.isInlineContiguousArraylet(array)) {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForContiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
-				} else {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForDiscontiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize())));
+		try {
+			if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
+				if (!_javaVM.isIndexableDataAddrPresent().isZero()) {
+					J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
+					UDATA dataSizeInBytes = ObjectModel.getDataSizeInBytes(array);
+					VoidPointer dataAddr = J9IndexableObjectHelper.getDataAddrForIndexable(array);
+					boolean isCorrectDataAddrPointer;
+					if (dataSizeInBytes.isZero()) {
+						isCorrectDataAddrPointer = (dataAddr.isNull() || dataAddr.equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize()))));
+					} else if (dataSizeInBytes.lt(_javaVM.arrayletLeafSize())) {
+						isCorrectDataAddrPointer = dataAddr.equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
+					} else {
+						isCorrectDataAddrPointer = (dataAddr.isNull());
+					}
+					if (false == isCorrectDataAddrPointer) {
+						return J9MODRON_GCCHK_RC_INVALID_INDEXABLE_DATA_ADDRESS;
+					}
 				}
-			} catch (NoSuchFieldException e) {
+			}
+		} catch (NoSuchMethodError | NoSuchFieldException | CorruptDataException e) {
```

Comment:
I don't think we want to catch `CorruptDataException` here; it should be propagated to the caller.
Why would we want to catch `NoSuchMethodError`?

Reply:
Ok. I will remove the `CorruptDataException` catch.

As for the `NoSuchMethodError` catch, there was a case relating to `isIndexableDataAddrPresent` during testing that came up. However, this escapes me now and I can't reproduce it, so I will remove the `NoSuchMethodError` catch as well.

Reply:
`NoSuchMethodError` might happen without the addition of `J9JavaVM.isIndexableDataAddrPresent` to `AuxFieldInfo29.dat`.